libraries/x86/test/test_x86_cmd.c

```diff
@@ -0,0 +1,51 @@
+#include <stdbool.h>
+#include <string.h>
+#include "log.h"
+#include "unity.h"
+#include "x86_cmd.h"
+
+static const char s_cmd[30];
+static size_t s_num_args;
+
+static void prv_handler(int client_fd, const char *cmd, const char *args[], size_t num_args,
+                        void *context) {
+  bool *received = context;
+  LOG_DEBUG("Handling cmd %s (%d args) from %d\n", cmd, num_args, client_fd);
+  for (size_t i = 0; i < num_args; i++) {
+    LOG_DEBUG("Arg %d: %s\n", i, args[i]);
+  }
+
+  const char *msg = "Response\n";
+  x86_socket_write(client_fd, msg, strlen(msg));
+
+  strncpy(s_cmd, cmd, sizeof(s_cmd));
+  s_num_args = num_args;
+  *received = true;
+}
+
+void setup_test(void) {
+  memset(s_cmd, 0, sizeof(s_cmd));
+  s_num_args = 0;
+}
+
+void teardown_test(void) {}
+
+void test_x86_cmd_client(void) {
+  volatile bool received = false;
+  x86_cmd_register_handler("test", prv_handler, &received);
+
+  LOG_DEBUG("Hopefully delaying a bit for the RX server to start\n");
```

Comment:
Nit: "Starting RX server" is a clearer message.

Reply:
It's not actually starting the RX server. The point is to delay the client until the RX server has probably started.

Reply:
Ah, ok.